tgui/packages/tgui/interfaces/MfdPanels/TargetAquisition.tsx

```diff
@@ -0,0 +1,442 @@
+      return [
+        { children: 'CANCEL', onClick: () => setLeftButtonMode(undefined) },
+        {
+          children: 'NORTH',
+          onClick: () => {
+            setFmOffsetDir('NORTH');
+            setLeftButtonMode(undefined);
+          },
+        },
+        {
+          children: 'SOUTH',
+          onClick: () => {
+            setFmOffsetDir('SOUTH');
+            setLeftButtonMode(undefined);
+          },
+        },
+        {
+          children: 'EAST',
+          onClick: () => {
+            setFmOffsetDir('EAST');
+            setLeftButtonMode(undefined);
+          },
+        },
+        {
+          children: 'WEST',
+          onClick: () => {
+            setFmOffsetDir('WEST');
+            setLeftButtonMode(undefined);
+          },
+        },
+      ];
+    }
+    return [];
+  };
+
+  const strikeConfigLabel =
+    strikeMode === 'weapon'
+      ? data.equipment_data.find((x) => x.mount_point === weaponSelected)?.name
+      : firemissionSelected !== undefined
+        ? data.firemission_data.find(
+          (x) => x.mission_tag === firemissionSelected
+        )?.name
+        : 'NONE';
+
+  const lazeIndex = lazes.findIndex((x) => x?.target_tag === selectedTarget);
+  const strikeReady = strikeMode !== undefined && lazeIndex !== -1;
+
+  const [targetOffset, setTargetOffset] = useLocalState(
+    context,
+    `${props.panelStateId}_targetOffset`,
+    0
+  );
+
+  const targets = range(targetOffset, targetOffset + 5).map(lazeMapper);
+  // lazes.map((x, i) => lazeMapper(i))
+  return (
+    <MfdPanel
+      panelStateId={props.panelStateId}
+      topButtons={[
+        {
+          children: 'FIRE',
+          onClick: () => {
+            if (strikeMode === undefined) {
+              return;
+            }
+            if (strikeMode === 'firemission') {
+              act('firemission-execute', {
+                tag: firemissionSelected,
+                direction: strikeDirection
+                  ? directionLookup[strikeDirection]
+                  : 1,
+                target_id: selectedTarget,
+                offset_direction: dirMap(fmOffsetDir),
+                offset_value: fmOffsetValue,
+              });
+            }
+            if (strikeMode === 'weapon') {
+              act('fire-weapon', { eqp_tag: weaponSelected });
+            }
+          },
+        },
+        {},
+        {},
+        {},
+        {
+          children: <Icon name="arrow-up" />,
+          onClick: () => {
+            if (targetOffset > 0) {
+              setTargetOffset(targetOffset - 1);
+            }
+          },
+        },
+      ]}
+      bottomButtons={[
+        {
+          children: 'EXIT',
+          onClick: () => setPanelState(''),
+        },
+        {},
+        {},
+        {},
+        {
+          children: <Icon name="arrow-down" />,
+          onClick: () => {
+            if (targetOffset < lazes.length - 1) {
+              setTargetOffset(targetOffset + 1);
```

Comment:
This should instead try to keep 5 good targets available rather than allowing you to scroll past when you don't need to Also, the arrows probably only need to appear at all if there are more than 5 targets.

![example](https://github.com/cmss13-devs/cmss13/assets/76988376/9ae9d83a-381b-4ca9-83d3-96b48df057e0)

Also would be nice if you could reuse this graphic you used in the weapons panel here again:
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/42a79607-b92e-4d3b-8aee-38a0f6c99128)

Reply:
![image](https://github.com/cmss13-devs/cmss13/assets/101871009/34c7d477-826a-45bf-b747-9130fd98b070)
I think the issues have been resolved. I've added some text to say how many targets there are as well.